### ISSUE TITLE
Sound when digging blocks (voxel-mine integration)

### ISSUE THREAD

**Issue by deathcap** [voxel/voxel-sfx#1]
While the block is being mined, and when when it breaks

https://github.com/deathcap/voxel-mine - might need new events

edit: also when tools break

**Comment by z3t0** [comment]
Do tools break at the moment?

**Comment by z3t0** [comment]
```js
mine.on('break', function(target) {
  // do something to this voxel (remove it, etc.)
});
```
Can be used to play a break sound, which could be generic. Or do we want some unique ones too, like glass breaking.

**Comment by z3t0** [comment]
Which plugin has the tools?  Should the functionality be added to those other plugins and make the voxel-sfx more generic to simply play sounds that we give it.

**Comment by deathcap** [comment]
> Do tools break at the moment?

Yah, they break when the damage reaches the maxDamage voxel-registry property, checked in https://github.com/deathcap/voxel-harvest

>  Or do we want some unique ones too, like glass breaking.

Sounds good, I like the idea you mentioned in https://github.com/deathcap/voxel-sfx/issues/6 about adding a sound property to the registry. If missing it could default to a generic sound.

> Which plugin has the tools? Should the functionality be added to those other plugins and make the voxel-sfx more generic to simply play sounds that we give it.

https://github.com/deathcap/voxel-pickaxe
https://github.com/deathcap/voxel-hammer

The advantage of isolating the sounds in voxel-sfx is that the user can enable/disable the plugin to globally enable/disable sound effects (this is supported in the UI with https://github.com/deathcap/voxel-plugins-ui).

**Comment by z3t0** [comment]
> The advantage of isolating the sounds in voxel-sfx is that the user can enable/disable the plugin
True, but I was thinking it would be a lot easier if we could put the adding blocks part in its own place instead of having it all over the place.

**Comment by z3t0** [comment]
Is there any way to determine the block broken from
```js
this.mine.on('break', function(target){
```
because then we can search it up on the registry for the sound

**Comment by z3t0** [comment]
Will this work?
```js
blockName = _this.registry.getBlockName(target.value);
```

**Comment by z3t0** [comment]
I have voxel-sfx ready to play block break sounds but now I need to figure out how to add sounds to registration, @deathcap could you take a look there?

**Comment by z3t0** [comment]
Where is the grass block registered? In which plugin?

**Comment by z3t0** [comment]
I am working on the registry plugin and have almost solved the whole thing only question is how do i access the props for a block. 

I know that most blocks use creativeTab : ' '
but how is that value accessed

**Comment by z3t0** [comment]
stupid question found the answer
```js
getProp(itemName, propName, arg)
```

**Comment by z3t0** [comment]
@deathcap You did a wonderful job in the registry.... its amazing

**Comment by z3t0** [comment]
ok block breaksound should be working now.
Just need to add `breakSound` to props in registration

**Comment by z3t0** [comment]
Also, if the block does not have breakSound, then the default break sound will happen

**Comment by deathcap** [comment]
Nice. For the record: https://github.com/deathcap/voxel-sfx/pull/10

**Comment by z3t0** [comment]
Right lol  that was by me.
